plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/contributors/keywords/ReturnKeywordHandler.kt

```diff
@@ -54,14 +61,23 @@ internal object ReturnKeywordHandler : CompletionKeywordHandler<KtAnalysisSessio
         return result
     }
 
-    private fun KtAnalysisSession.addAllReturnVariants(result: MutableList<LookupElement>, returnType: KtType, label: Name?) {
+    private fun KtAnalysisSession.addAllReturnVariants(
+        result: MutableList<LookupElement>,
+        returnType: KtType,
+        label: Name?,
+        isLikelyInPositionForReturn: Boolean = false
+    ) {
         val isUnit = returnType.isUnit
-        result.add(createKeywordElementWithSpace("return", tail = label?.labelNameToTail().orEmpty(), addSpaceAfter = !isUnit))
+        result.add(createKeywordElementWithSpace("return", tail = label?.labelNameToTail().orEmpty(), addSpaceAfter = !isUnit).also {
+            it.isReturnAtHighlyLikelyPosition = isLikelyInPositionForReturn
+        })
         getExpressionsToReturnByType(returnType).mapTo(result) { returnTarget ->
             if (label != null || returnTarget.addToLookupElementTail) {
```

Comment:
SUGGESTION: Here I would extract the result of `if` to a variable, since the construction `if (...) {...} else {...}.also {...}` looks a little bit fishy